sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -891,6 +1013,13 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                 {
                     await receiver.AbandonAsync(message.LockToken).ConfigureAwait(false);
                 }
+
+                // we need to propagate this in order to dispose the session receiver
+                // in the same place where we are creating them.
+                if ((ex as ServiceBusException)?.Reason == ServiceBusException.FailureReason.SessionLockLost)
```

Comment:
Should this go at the top of the catch block? Otherwise I think we would be calling the user error handler twice (once here and once at the higher level catch block).

Reply:
Good catch!